lib/test.rb

```diff
@@ -657,6 +671,12 @@ def formula(formula_name)
       @formulae_that_have_been_built << formula_name
 
       formula = Formulary.factory(formula_name)
+      if formula.deprecated? || formula.disabled?
+        verb = formula.deprecated? ? :deprecated : :disabled
+        puts "#{formula.full_name} has been #{verb}!"
+        skip formula.name
+        return
+      end
```

Comment:
It makes sense to skip as a dependent but do we actually want to skip it if changes to that formula are the main focus of the pull request being tested?

Reply:
I think so. Definitely if they are disabled but even when deprecated I wonder what the merits are of continuing to run CI jobs on them. CC @sjackman @SMillerDev @reitermarkus for thoughts

Reply:
@Bo98 Let me know what you think of this now.